xamarin/Xamarin.Forms
# [iOS|Android] Fix screen reader order

### Description of Change ###

* iOS MasterDetailPage will no longer read MasterPage contents when the master page is hidden and will no longer read DetailPage contents when the master page is visible.
* iOS can now tab stop on a Frame and can read its contents properly.
* iOS/Android screen readers should no longer skip over layouts and should read all children in declaration order and/or TabIndex order.
* TabIndex should no longer return elements that are `Enabled = false` or `IsAccessibilityElement = false`.

### Issues Resolved ### 


- fixes #9653
- fixes #4220 
- fixes #9191 
- fixes #8613
- fixes #9137
- fixes #8691 
- fixes #7606
- fixes #4270
- fixes #5879

### API Changes ###
 
These are breaking API changes, but they drive me crazy. I will revert if someone has an issue with this.

`TabIndexExtensions`:
```diff
-public static SortedDictionary<int, List<ITabStopElement>> GetSortedTabIndexesOnParentPage(this VisualElement element, out int countChildrensWithTabStopWithoutThis)
+public static SortedDictionary<int, List<ITabStopElement>> GetSortedTabIndexesOnParentPage(this VisualElement element)


-public static IDictionary<int, List<ITabStopElement>> GetTabIndexesOnParentPage(this ITabStopElement element, out int countChildrensWithTabStopWithoutThis, bool checkContainsElement = true)
+public static IDictionary<int, List<ITabStopElement>> GetTabIndexesOnParentPage(this ITabStopElement element, out int countChildrenWithTabStopWithoutThis)
```

### Platforms Affected ### 

- Core/XAML (all platforms)
- iOS
- Android
- UWP

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

1. Use TalkBack/VoiceOver to test the Control Gallery and make sure everything is read in an expected order.
2. Use keyboard to test that TabIndex is still functioning as expected.

### PR Checklist ###
